src/DI/ElasticsearchExtension.php

```diff
@@ -5,29 +5,40 @@
 use Elasticsearch\Client;
 use Elasticsearch\ClientBuilder;
 use Nette\DI\CompilerExtension;
-
-class ElasticsearchExtension extends CompilerExtension
+use Nette\DI\Helpers;
+use Nette\Schema\Expect;
+use Nette\Schema\Schema;
+use stdClass;
+
+/**
+ * @property-read stdClass $config
+ */
+final class ElasticsearchExtension extends CompilerExtension
 {
 
-	/** @var mixed[] */
-	private $defaults = [
-		'hosts' => ['localhost'],
-	];
+	public function getConfigSchema(): Schema
+	{
+		return Expect::structure([
+			'hosts' => Expect::arrayOf('string')->required(),
+		]);
+	}
 
 	public function beforeCompile(): void
 	{
-		if (!isset($this->config['hosts'])) {
-			$this->defaults['hosts'] = [];
-		}
+		$builder = $this->getContainerBuilder();
+		$config = $this->config;
 
-		$config = $this->getConfig();
+		// Default values cannot be in schema, arrays are merged by keys
+		if ($config->hosts === []) {
+			$config->purge = Helpers::expand(['localhost'], $builder->parameters);
```

Comment:
Expand is not needed for value without placeholders.
Also, it's `hosts`, not `purge`